ReactWindows/ReactNative/Modules/Clipboard/ClipboardModule.cs

```diff
@@ -0,0 +1,88 @@
+﻿using ReactNative.Bridge;
+using System;
+using DT = Windows.ApplicationModel.DataTransfer;
+
+namespace ReactNative.Modules.Clipboard
+{
+    /**
+    * A module that allows JS to get/set clipboard contents.
+    */
+    class ClipboardModule : ReactContextNativeModuleBase
+    {
+        /// <summary>
+        /// Instantiates the <see cref="ClipboardModule"/>.
+        /// </summary>
+        /// <param name="reactContext">The context.</param>
+        internal ClipboardModule(ReactContext reactContext)
+            : base(reactContext)
+        {
+        }
+
+        /// <summary>
+        /// The name of the native module.
+        /// </summary>
+        public override string Name
+        {
+            get
+            {
+                return "Clipboard";
+            }
+        }
+
+        [ReactMethod]
+        public void getString(IPromise promise)
+        {
+            if (promise == null)
+            {
+                throw new ArgumentNullException(nameof(promise));
+            }
+
+            DispatcherHelpers.RunOnDispatcher(async () =>
+            {
+                try
+                {
+                    var clip = DT.Clipboard.GetContent();
+                    if (clip == null)
+                    {
+                        promise.Resolve("");
+                    }
+                    else if (clip.Contains(DT.StandardDataFormats.Text))
+                    {
+                        string text = await clip.GetTextAsync();
+                        promise.Resolve(text);
+                    }
+                    else
+                    {
+                        promise.Resolve("");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    promise.Reject(ex);
+                }
+            });
+        }
+
+        [ReactMethod]
+        public void setString(string text)
+        {
+            DispatcherHelpers.RunOnDispatcher(() =>
+            {
+                try
+                {
+                    if (text == null)
+                    {
+                        DT.Clipboard.Clear();
+                    }
+                    else
+                    {
+                        DT.DataPackage package = new DT.DataPackage();
+                        package.SetData(DT.StandardDataFormats.Text, text);
+                        DT.Clipboard.SetContent(package);
+                    }
+                }
+                catch (Exception) {}

```

Comment:
Actually, yes, for sure don't catch it.  The exception thrown here should be propagated to a RedBox exception if thrown, which would allow the user to take action on it.

---
In reply to: [55373480](https://github.com/CatalystCode/react-native/pull/233#discussion_r55373480) [](ancestors = 55373480)